src/Microsoft.DotNet.XHarness.Common/Execution/ProcessManager.cs

```diff
@@ -318,8 +318,17 @@ public abstract class ProcessManager : IProcessManager
                     }
                 }
             }
+            else
+            {
+                await WaitForExitAsync(process);
+            }
```

Comment:
this isn't related to the PR but seems like an unrelated issue: if we have a timeout and WaitForExitAsync returns false then we'd call it a second time